From b33b618b03579438efceeda51c2d1c6eab0ddea9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Markus=20M=C3=BCtzel?= <markus.muetzel@gmx.de>
Date: Sun, 28 Nov 2021 21:29:13 +0100
Subject: [PATCH] Fix compilation with Octave 8 (bug #61577).

* src/Mshichmanhodgesmosfet.cc (FMshichmanhodgesmosfet): Remove usage of
`error_state`. It was unconditionally set to 0 since about 6 years ago and
will finally be removed in Octave 8.
---
 src/Mshichmanhodgesmosfet.cc | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/Mshichmanhodgesmosfet.cc b/src/Mshichmanhodgesmosfet.cc
index 4b3e51a..1c0d7c6 100644
--- src/Mshichmanhodgesmosfet.cc
+++ src/Mshichmanhodgesmosfet.cc
@@ -342,10 +342,8 @@ the output structures.\n\
     error("Mshichmanhodgesmosfet: double type value expected as time instant.\n");
   }
 
-  if (!error_state)
-    {
-      //FIXME: create enum of cases and use switch?
-      if (eltype == "NMOS")
+  //FIXME: create enum of cases and use switch?
+  if (eltype == "NMOS")
 	{
 	  //FIXME: change parameters to a single map or Octave_map
 	  /* Default n-MOS parameters*/
@@ -543,7 +541,5 @@ the output structures.\n\
       else
 	error("Mshichmanhodgesmosfet: unknown element type.\n");
 
-    }
-
   return (retval);
 }
-- 
2.34.1.windows.1

